tethys_apps/cli/scaffold_templates/app_templates/default/tethysapp/+project+/app.py_tmpl

```diff
@@ -16,6 +18,11 @@ class {{class_name}}(TethysAppBase):
     tags = '{{tags}}'
     enable_feedback = False
     feedback_emails = []
+    analytics = bool('analytical' in settings.INSTALLED_APPS and settings.GOOGLE_ANALYTICS_JS_PROPERTY_ID != '')
+
+    with open(os.path.join(os.path.dirname(__file__), 'templates/{{project}}/analytics.html'), 'w') as file:
```

Comment:
It's not a good idea to have logic within the class declaration. This really should be moved into a method, perhaps the constructor? Why not just add the analytics html file as another template file that is generated when the app is created from scaffold?

Reply:
Also, what does this page do? Does it show stats for the entire portal (not just this app)? If so, this should probably be a page that is added to the admin pages as a portal-wide status page accessible only by portal admins.

Reply:
Good question, this is the most substantive change to the code. django analytical makes implementing a variety of tracking services into a django app easier by letting you use some standard tags and lets the package implement the scripts for each service customized for your IDs. The bool check is looking to see if the portal admins have specified the tracking id for their accounts in the appropriate settings file. If they have, then the analytics.html page gets filled with the tags to implement tracking. If not, then the page is wiped when the file is opened so no tags get added to the scripts section that would break the app (because the package and/or tracking id haven't been specified). I agree it could be moved somewhere else so I'm behind any suggestion for a different method to put it in.